### ISSUE TITLE
VM hanging on 100% after live-migration

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1042]
Originally reported of Google Code with ID 986.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".

<b>What distribution are you using?</b>
# gnt-cluster --version
gnt-cluster (ganeti v2.11.6) 2.11.6
# gnt-cluster version
Software version: 2.11.6
Internode protocol: 2110000
Configuration format: 2110000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.11.6
# hspace --version
hspace (ganeti) version v2.11.6
compiled with ghc 7.4
running on linux x86_64
# cat /etc/debian_version
7.7
# apt-cache policy ganeti
ganeti:
  Installed: 2.11.6-1~bpo70+1
  Candidate: 2.11.6-1~bpo70+1
  Package pin: 2.11.6-1~bpo70+1
  Version table:
 *** 2.11.6-1~bpo70+1 990
        100 http://debian.xxxxxxxx.de/debian/ wheezy-backports/main amd64 Packages
        100 /var/lib/dpkg/status
# kvm --version
QEMU emulator version 1.1.2 (qemu-kvm-1.1.2+dfsg-6+deb7u5, Debian), Copyright (c) 2003-2008 Fabrice Bellard


<b>What steps will reproduce the problem?</b>
1. gnt-instance migrate [vm]


<b>What is the expected output? What do you see instead?</b>
in 1/4 - 1/5 of the migration: a cpu load of 75-100% per VCPU of the VM


<b>Please provide any additional information below.</b>
- the console (gnt-instance console doesn't respond)
- spice gives a black screen (sending any keys won't do anything, inc. Ctrl + Alt + Del)
- this happens about on 3-4 VMs of ~ 16 VMs every maintenance
- the VMs aren't the same
- exept for once it was always a debian Guest (the other one time it was CentOS 6)
- at about 80% of the time it is a debian-stable Guest (60% of the guest are running on stable)
- i had this problem with the qemu-cpu and with host-cpu for the guest
- it happens to multi-vcpu VMs as to single-vcpu VMs and the amount of RAM doesn't seam to have an effect (I found a mail from Juli 2013 referring to this)


output of ps aux | grep [vm]
root     20885 75.2  6.1 2485104 2031276 ?     Sl   01:58   1:00 /usr/bin/kvm -name zzzzzzz.yyyyyy.xxxxxxxxx.de -m 2048 -smp 1 -pidfile /var/run/ganeti/kvm-hypervisor/pid/zzzzzzz.yyyyyy.xxxxxxxxx.de -balloon virtio,id=balloon,bus=pci.0,addr=0x3 -daemonize -machine pc-1.1 -
monitor unix:/var/run/ganeti/kvm-hypervisor/ctrl/zzzzzzz.yyyyyy.xxxxxxxxx.de.monitor,server,nowait -serial unix:/var/run/ganeti/kvm-hypervisor/ctrl/zzzzzzz.yyyyyy.xxxxxxxxx.de.serial,server,nowait -usb -device virtio-serial-pci,id=spice,bus=pci.0,addr=0x4 -device virtseria
lport,chardev=spicechannel0,name=com.redhat.spice.0 -chardev spicevmc,id=spicechannel0,name=vdagent -spice addr=0.0.0.0,port=11026,disable-ticketing -cpu host -vga qxl -uuid 3d7c557e-f411-44c9-bed4-f41f63c84296 -netdev type=tap,id=hotnic-6ddd16f2-pci-7,fd=8 -device virtio-
net-pci,mac=00:e6:e7:1d:ff:d3,id=hotnic-6ddd16f2-pci-7,bus=pci.0,addr=0x7,netdev=hotnic-6ddd16f2-pci-7 -incoming tcp:10.165.94.2:8102 -qmp unix:/var/run/ganeti/kvm-hypervisor/ctrl/zzzzzzz.yyyyyy.xxxxxxxxx.de.qmp,server,nowait -qmp unix:/var/run/ganeti/kvm-hypervisor/ctrl/z
zzzzzz.yyyyyy.xxxxxxxxx.de.kvmd,server,nowait -boot c -device virtio-blk-pci,drive=hotdisk-ebfce63c-pci-5,id=hotdisk-ebfce63c-pci-5,bus=pci.0,addr=0x5 -drive file=/var/run/ganeti/instance-disks/zzzzzzz.yyyyyy.xxxxxxxxx.de:0,format=raw,if=none,cache=writeback,id=hotdisk-ebf
ce63c-pci-5,bus=0,unit=5 -device virtio-blk-pci,drive=hotdisk-cde35ef8-pci-6,id=hotdisk-cde35ef8-pci-6,bus=pci.0,addr=0x6 -drive file=/var/run/ganeti/instance-disks/zzzzzzz.yyyyyy.xxxxxxxxx.de:1,format=raw,if=none,cache=writeback,id=hotdisk-cde35ef8-pci-6,bus=0,unit=6
```
Originally added on 2014-11-07 02:20:21 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I have upgraded to 2.1 (from debian backports), but this issue still exists
but I have only tested this with one VM
```
Originally added on 2015-02-10 11:19:33 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I have upgraded to KVM 2.1 (from debian backports), but this issue still exists
but I have only tested this with one VM
```
Originally added on 2015-02-10 11:35:06 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
# sources
deb http://security.debian.org/                 wheezy/updates  main contrib non-free
deb http://debian.xxxxxxxx.de/debian            wheezy          main contrib non-free
deb http://debian.xxxxxxxx.de/debian            wheezy-backports main contrib


# preferences
Package: ganeti*
Pin: release n=wheezy-backports
Pin-Priority: 990

Package: drbd8-utils drbd-utils linux-image-amd64 initramfs-tools
Pin: release n=wheezy-backports
Pin-Priority: 990

Package: qemu-* libspice-server1 libusb-1.0-0 libusbredirparser1 libxen-4.1 libxenstore3.0 ipxe-qemu
Pin: release n=wheezy-backports
Pin-Priority: 990


# dpkg -l | grep qemu
ii  ipxe-qemu                               1.0.0+git-20131111.c3d1e78-2.1~bpo70+1 all          PXE boot firmware - ROM images for qemu
ii  qemu-kvm                                2.1+dfsg-9~bpo70+1                     amd64        QEMU Full virtualization on x86 hardware
ii  qemu-system-common                      2.1+dfsg-9~bpo70+1                     amd64        QEMU full system emulation binaries (common files)
ii  qemu-system-x86                         2.1+dfsg-9~bpo70+1                     amd64        QEMU full system emulation binaries (x86)
ii  qemu-utils                              2.1+dfsg-9~bpo70+1                     amd64        QEMU utilities

```
Originally added on 2015-02-10 11:49:34 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I debuged this a little more and found out that the monitor socket works,
I could get details like "info registers" (which were changing)
but creating screenshots would only give a black image
and "log [all]" wouldn't print anything into the logfiles (I even enabled this on VMs which crashed before they migrated)

The only thing i noticed is that all other VMs had a "(halted)" in there "info cpus" (I suppose this means that the VM was atm. idle)
# hanging VM
(qemu) QEMU 2.1.2 monitor - type 'help' for more information
(qemu) info cpus
* CPU #0: pc=0xffffffff81036228 thread_id=31975
  CPU #1: pc=0xffffffff8103622f thread_id=31976
  CPU #2: pc=0xffffffff81036228 thread_id=31977
  CPU #3: pc=0xffffffff81036239 thread_id=31978
# running VM
(qemu) QEMU 2.1.2 monitor - type 'help' for more information
(qemu) info cpus
* CPU #0: pc=0xffffffff8102b36c (halted) thread_id=30967
  CPU #1: pc=0xffffffff8102b36c (halted) thread_id=30968
```
Originally added on 2015-02-20 18:47:16 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2015-05-20 12:29:54 +0000 UTC.

Changed State: Duplicate
Marked as duplicate of #1106

**Comment by Ganeti-Issues-Migrator** [comment]
```
there were some mails on the mailinglist about this lately, and it seams that disabeling the cpu feature kvm_steal_time on the client fixes this.

On Wed, Jun 3, 2015 at 11:21 AM, Apollon Oikonomopoulos <apollon@skroutz.gr> wrote:
> Hi,
>
> It seems this is a bug with steal time accounting. I don't know (yet) if
> it's qemu, the kernel or both, but if you turn off kvm steal time
> reporting, it should go away:
>
> gnt-instance modify -H 'cpu_type=qemu64,-kvm_steal_time' <instance_name>
>
> Can you give it a try?
>
> Regards,
> Apollon


```
Originally added on 2015-06-11 15:51:08 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
As Ganeti already has all the features needed to remedy this until the root cause is found, I'll mark the issue as WontFix.
```
Originally added on 2015-06-11 15:58:38 +0000 UTC.

Changed State: WontFix
No longer marked as duplicate.{false 986 As Ganeti already has all the features needed to remedy this until the root cause is found, I'll mark the issue as WontFix. 2015-06-11 15:58:38 +0000 UTC 0 [] map[] [] [] WontFix   map[] map[] map[] map[] 0 true}

**Comment by Ganeti-Issues-Migrator** [comment]
```
well I'd guess this should rather be handled by debian in this case.

I'll see what Apollon says to that. 
```
Originally added on 2015-06-11 16:01:37 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I think this is not the same as the steal time issue. With the steal time issue the guest continues to work but does not report CPU usage correctly.

I think this is the TSC issue (https://bugs.debian.org/786789) which has been fixed in qemu 2.1.3. Debian Jessie is shipping 2.1.2 right now, but we hope it will be updated to 2.1.3 in a point release. This is also probably the same as issue #1050 and both of them can't be marked as WontFix because there's nothing Ganeti can actually do for them.
```
Originally added on 2015-06-12 09:10:50 +0000 UTC.
